hazelcast/src/main/java/com/hazelcast/topic/impl/reliable/MessageRunner.java

```diff
@@ -196,34 +199,28 @@ private boolean handleOperationTimeoutException() {
     protected abstract Throwable adjustThrowable(Throwable t);
 
     /**
-     * Handles a {@link StaleSequenceException} associated with requesting
-     * a sequence older than the {@code headSequence}.
-     * This may indicate that the reader was too slow and items in the
-     * ringbuffer were already overwritten.
+     * Handles a sudden jump in received sequence numbers. This may indicate
+     * that the reader was too slow and items in the ringbuffer were already overwritten.
      *
-     * @param staleSequenceException the exception
-     * @return if the exception was handled and the listener may continue reading
+     * @param newSequence the new, unexpected sequence number encountered
+     * @return if the listener may continue reading
      */
-    private boolean handleStaleSequenceException(StaleSequenceException staleSequenceException) {
-        long headSeq = getHeadSequence(staleSequenceException);
+    private boolean handleSilentSequenceJump(long newSequence) {
         if (listener.isLossTolerant()) {
             if (logger.isFinestEnabled()) {
-                logger.finest("MessageListener " + listener + " on topic: " + topicName + " ran into a stale sequence. "
-                        + "Jumping from oldSequence: " + sequence
-                        + " to sequence: " + headSeq);
+                logger.finest("MessageListener " + listener + " on topic: " + topicName + " ran into a silent sequence"
+                        + " jump from oldSequence: " + sequence + " to sequence: " + newSequence);
             }
-            sequence = headSeq;
+            sequence = newSequence;
```

Comment:
We will assign this once the method returns so you can remove this.

Reply:
Ok, good point, will do.